### ISSUE TITLE
failed to start tmux because of 'missing or unsuitable terminal' error

### ISSUE THREAD

**Issue by uesyn** [pkgxdev/pantry#3809]
After [Don’t set TERMINFO by mxcl · Pull Request #3792 · pkgxdev/pantry](https://github.com/pkgxdev/pantry/pull/3792) was merged, we couldn't run tmux  in Linux because of `missing or unsuitable terminal` error.

**How to reproduce**
We can reproduce easily it with docker.

```sh
$ docker run -it pkgxdev/pkgx
pkgx $ pkgx --sync

pkgx $ pkgx +tmux@3.3a tmux
 ✓  ~/.pkgx/curl.se/ca-certs/v2023.8.22
 ✓  ~/.pkgx/github.com/tmux/tmux/v3.3a
 ✓  ~/.pkgx/libevent.org/v2.1.12
 ✓  ~/.pkgx/invisible-island.net/ncurses/v6.4.0
 ✓  ~/.pkgx/openssl.org/v1.1.1w
missing or unsuitable terminal: xterm
```

It seems that TERMINFO_DIRS is not found.

```sh
pkgx $ eval "$(pkgx --internal.use +tmux)"
pkgx $ echo $TERMINFO_DIRS
/root/.pkgx/invisible-island.net/ncurses/v6.4.0/share/terminfo
pkgx $ ls /root/.pkgx/invisible-island.net/ncurses/v6.4.0/share/terminfo
ls: cannot access '/root/.pkgx/invisible-island.net/ncurses/v6.4.0/share/terminfo': No such file or directory
```

In my macOS env, TERMINFO_DIRS was found. I don't know why TERMINFO_DIR directory was not generated by ncurses in Linux.

**Comment by mxcl** [comment]
k well, apparently the linux build doesn't generate any `terminfo` but the mac build does. Will require some deep dives to fix. sigh

**Comment by mxcl** [comment]
Using the above instructions unsetting `TERMINFO_DIRS` doesn't fix it. I’m not very familiar with this stuff.

I'm confused how it worked before #3792

**Comment by uesyn** [comment]
@mxcl 

I don't know why terminfo db was not generated in linux env, but I could generate it with below  the change.

https://github.com/uesyn/pantry/commit/96942fb7af309347ee9e8eb1446814b0f2be5adb

As far as I confirmed, the change works well in MacOS(arm64) and Linux(amd64) env.

I think we don't need `--with-default-terminfo-dir` whitch is used by configure flag, because this flag value is final default value in case that paths set in TERMINFO_DIRS(we use) or TERMINFO is not found. 
ref: https://invisible-island.net/ncurses/man/ncurses.3x.html#h3-TERMINFO

TERMINFO_DIRS can be set multiple paths. Instead of `--with-default-terminfo-dir`, I added '/usr/share/terminfo' to TERMINFO_DIRS.

If looks good to you, I can PR it.
